src/cascade/model/session.py

```diff
@@ -0,0 +1,552 @@
+            with_density = with_density.assign(name=with_density.index.astype(str))
+        elif not with_density.name.isnull().empty:
+            raise RuntimeError(f"There are some data values that lack data names.")
+        else:
+            pass  # There are data names everywhere.
+        if "hold_out" not in with_density.columns:
+            with_density = with_density.assign(hold_out=0)
+        for additional in ["nu", "eta"]:
+            if additional not in with_density.columns:
+                with_density = with_density.assign(**{additional: nan})
+        for expand in ["age", "time"]:
+            point_dimension = (f"{expand}_lower" not in with_density.columns and
+                               f"{expand}_upper" not in with_density.columns)
+            if point_dimension and expand in with_density.columns:
+                with_density = with_density.assign(
+                    **{f"{expand}_lower": with_density[expand], f"{expand}_upper": with_density[expand]})
+                with_density = with_density.drop(columns=[expand])
+
+        self.dismod_file.data = with_density.rename(columns={"mean": "meas_value", "std": "meas_std",
+                                                             "name": "data_name"})
+
+    def get_predict(self):
+        avgint = self.read_avgint()
+        raw = self.dismod_file.predict.merge(avgint, on="avgint_id", how="left")
+        not_predicted = avgint[~avgint.avgint_id.isin(raw.avgint_id)]
+        return raw.drop(columns=["avgint_id"]), not_predicted.drop(columns=["avgint_id"])
+
+    def _basic_db_setup(self):
+        """These things are true for all databases."""
+        # Density table does not depend on model.
+        self.dismod_file.density = pd.DataFrame({"density_name": [x.name for x in DensityEnum]})
+
+        # Standard integrand naming scheme.
+        all_integrands = default_integrand_names()
+        self.dismod_file.integrand = all_integrands
+        # Fill in the min_meas_cv later if required. Ensure integrand kinds have
+        # known IDs early. Not nan because this "is non-negative and less than
+        # or equal to one."
+        self.dismod_file.integrand["minimum_meas_cv"] = 0
+
+        self.dismod_file.rate = pd.DataFrame(dict(
+            rate_id=[rate.value for rate in RateEnum],  # Will be 0-4.
+            rate_name=[rate.name for rate in RateEnum],
+            parent_smooth_id=nan,
+            child_smooth_id=nan,
+            child_nslist_id=nan,
+        ))
+
+        # Defaults, empty, b/c Brad makes them empty even if there are none.
+        for create_name in ["nslist", "nslist_pair", "mulcov", "smooth_grid", "smooth"]:
+            setattr(self.dismod_file, create_name, self.dismod_file.empty_table(create_name))
+        self.dismod_file.log = make_log_table()
+        self._create_options_table()
+
+    def _create_options_table(self):
+        # Options in grey were rejected by Dismod-AT despite being in docs.
```

Comment:
In grey? Commented out? But I don't see any...

Reply:
I figured out what the options really are by working through all examples. Thanks for catching this.